src/components/Application/ProgressBarNEW.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import { Accordion, Container, Grid, Header, Icon, Label, List, Sticky } from 'semantic-ui-react'
+import strings from '../../utils/constants'
+import { FullStructure, PageNEW, Progress } from '../../utils/types'
+
+interface ProgressBarProps {
+  structure: FullStructure
+  changePage: (sectionCode: string, pageNumber: number) => void
+  current: {
+    sectionCode: string
+    page: number
+  }
+}
+
+const ProgressBarNEW: React.FC<ProgressBarProps> = ({ structure, changePage, current }) => {
+  const {
+    info: { isLinear },
+    sections,
+  } = structure
+
+  const activeIndex =
+    Object.values(sections).findIndex(({ details }) => details.code === current.sectionCode) || 0
+
+  const getPageList = (sectionCode: string, pages: { [pageName: string]: PageNEW }) => {
+    const isActivePage = (sectionCode: string, pageNumber: number) =>
+      current.sectionCode === sectionCode && current.page === pageNumber
+
+    return (
+      <List
+        link
+        style={{ paddingLeft: '50px' }}
+        items={Object.entries(pages).map(([pageName, { number, progress }]) => ({
+          key: `ProgressSection_${sectionCode}_${number}`,
+          active: isActivePage(sectionCode, number),
+          as: 'a',
+          icon: progress ? getIndicator(progress) : null,
+          content: pageName,
+          onClick: () => changePage(sectionCode, number),
+        }))}
+      />
+    )
+  }
+
+  const getIndicator = (progress: Progress, step?: number) => {
+    const { completed, valid } = progress
+    if (!completed)
+      return step ? (
+        <Label circular as="a" basic color="blue" key={`progress_${step}`}>
+          {step}
+        </Label>
+      ) : (
+        <Icon name="circle outline" />
+      )
+    return (
+      <Icon
+        name={valid ? 'check circle' : 'exclamation circle'}
+        color={valid ? 'green' : 'red'}
+        size={step ? 'large' : 'small'}
+      />
+    )
+  }
+
+  const isSectionDisabled = (index: number) => {
```

Comment:
I think we would use firstStrictInvalidPageFor this (if linear)

Reply:
Have discussed briefly with Carl, maybe we shouldn't disable sections/pages just because since they are still links when hoovering it display the icon as clickable, is just the label that is changing. And when the user clicks on the a page ahead (on linear applications) it should just run the validation and display errors on the current page. 

So changing now to have all pages/sections enabled. Seems good?

Reply:
yep